src/Analyzers/CSharp/Analyzers/OrderModifiers/CSharpOrderModifiersDiagnosticAnalyzer.cs

```diff
@@ -30,29 +30,22 @@ public CSharpOrderModifiersDiagnosticAnalyzer()
             ReportDiagnostic severity,
             SyntaxNode root)
         {
-            foreach (var child in root.ChildNodesAndTokens())
+            foreach (var node in root.ChildNodes())
             {
-                if (child.IsNode)
+                if (node is MemberDeclarationSyntax || node.IsKind(SyntaxKind.LocalFunctionStatement))
                 {
-                    var node = child.AsNode();
-                    if (node is MemberDeclarationSyntax memberDeclaration)
-                    {
-                        CheckModifiers(context, preferredOrder, severity, memberDeclaration);
-
-                        // Recurse and check children.  Note: we only do this if we're on an actual 
-                        // member declaration.  Once we hit something that isn't, we don't need to 
-                        // keep recursing.  This prevents us from actually entering things like method 
-                        // bodies.
-                        Recurse(context, preferredOrder, severity, node);
-                    }
-                    else if (node is AccessorListSyntax accessorList)
+                    CheckModifiers(context, preferredOrder, severity, node);
+                }
+                else if (node is AccessorListSyntax accessorList)
```

Comment:
weird.  i don't get why we handle this, instead of just handling AccessorDeclarationSyntax.

Reply:
@CyrusNajmabadi The previous code used to bail early on children of member declarations, in which case we won't see accessor declarations.

The new code can be changed to handle accessor declaration directly, but since you actually think there is a perf impact, let's keep this until we resolve the performance impact.